### PR TITLE
Link to Artificial Intelligence book Added

### PR DESCRIPTION
**Pull request by theayangupta**
## What does this PR do?
Add resource(s) | Remove resource(s) | Add info | Improve repo
Add resource

### Description
I have added the link to the book "Artificial Intelligence - Agents and Environments" authored by William John Teahan.

### Why is this valuable (or not)?
This book is the first in a series on Artificial Intelligence. It provides an introduction to the topic with an emphasis on the use of agent-oriented design. Topics include agents, environments, agent movement, and agent embodiment. It also provides an introduction to programming in NetLogo.

### How do we know it's really free?
It doesn't require any sign-up/login. The website is legal.

## Checklist:
- [✅ ] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [✅ ] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [✅ ] Include author(s) and platform where appropriate.
- [✅ ] Put lists in alphabetical order, correct spacing.
- [✅ ] Add needed indications (PDF, access notes, under construction).
- [✅ ] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-subjects.md [hunk 1]
```diff
@@ -112,6 +113,7 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 * [Essential Algorithms](https://www.programming-books.io/essential/algorithms) - Krzysztof Kowalczyk and  Stack Overflow Documentation project (HTML)
 * [Foundations of Computer Science](http://infolab.stanford.edu/~ullman/focs.html) - Al Aho, Jeff Ullman
 * [Geometry Algorithms](http://geomalgorithms.com) - Dan Sunday
+* [Introduction to Algorithms](https://sd.blackball.lv/library/Introduction_to_Algorithms_Third_Edition_(2009).pdf) - Thomas H. Cormen, Charles E. Leiserson, Ronald L. Rivest, Clifford Stein (PDF)
```
Comment (shrn01) [inline books/free-programming-books-subjects.md:116]: Not an authorized distribution

File: books/free-programming-books-subjects.md [hunk 2]
```diff
@@ -99,6 +99,7 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 * [Analysis and Design of Algorithms](http://www.cse.iitd.ernet.in/~ssen/csl356/admin356.html) - Sandeep Sen, IIT Delhi
 * [Animated Algorithm and Data Structure Visualization](http://visualgo.net) (Resource)
 * [Annotated Algorithms in Python: Applications in Physics, Biology, and Finance](https://github.com/mdipierro/nlib) - Massimo di Pierro
+* [Artificial Intelligence - Agents and Environments](http://www.e-booksdirectory.com/details.php?ebook=4457) - William John Teahan
```
Comment (shrn01) [inline books/free-programming-books-subjects.md:102]: Is this an authorized distribution?

Comment (theayangupta) [reply books/free-programming-books-subjects.md:102]: @shrn01 Yes, this is an authorized distribution as per my findings.

Comment (eshellman) [reply books/free-programming-books-subjects.md:102]: E-books directory is an aggregator site. notice that the links on the page you've supplied points to the Bookboon site. that means that we can't check to see if the book is still available. So we won't use links to ebook directory.
In other words, _it's not even a distribution_, let alone authorized!

**Pull request closed without merging**